devito/finite_differences/differentiable.py

```diff
@@ -1005,3 +1006,28 @@ def _diff2sympy(obj):
 evalf_table[Add] = evalf_table[sympy.Add]
 evalf_table[Mul] = evalf_table[sympy.Mul]
 evalf_table[Pow] = evalf_table[sympy.Pow]
+
+
+# Interpolation for finite differences
+@singledispatch
+def interp_for_fd(expr, x0, **kwargs):
+    return expr
+
+
+@interp_for_fd.register(AbstractFunction)
+def _(expr, x0, **kwargs):
+    from devito.finite_differences.derivative import Derivative
```

Comment:
Can this not be imported at the top?

Reply:
No circular import, why it's also imported inside `def diff` in Differentiable